op-service/client/backoff.go

```diff
@@ -0,0 +1,82 @@
+package client
+
+import (
+	"context"
+	"time"
+
+	"github.com/ethereum/go-ethereum"
+	"github.com/ethereum/go-ethereum/rpc"
+
+	"github.com/ethereum-optimism/optimism/op-service/backoff"
+)
+
+var (
+	// ExponentialBackoff is the default backoff strategy.
+	ExponentialBackoff = backoff.Exponential()
+)
+
+// InnerRPC is a minimal [client.RPC] interface that is used by the backoff client.
+//
+//go:generate mockery --name InnerRPC --output ./mocks/
+type InnerRPC interface {
+	Close()
+	CallContext(ctx context.Context, result any, method string, args ...any) error
+	BatchCallContext(ctx context.Context, b []rpc.BatchElem) error
+	EthSubscribe(ctx context.Context, channel any, args ...any) (ethereum.Subscription, error)
+}
+
+// backoffEthClient wraps a [InnerRPC] with a backoff strategy.
+type backoffEthClient struct {
+	c             InnerRPC
+	retryAttempts int
+	strategy      backoff.Strategy
+}
+
+// NewBackoffClient creates a new backoff client.
+// The backoff strategy is optional, if not provided, the default exponential backoff strategy is used.
+func NewBackoffClient(c InnerRPC, retries int, strategy ...backoff.Strategy) *backoffEthClient {
```

Comment:
nit: I'd be tempted to call this `NewRetryingClient` since the key functionality it adds is the ability to retry. It just happens to use a backoff strategy as part of retrying.

Reply:
That's good with me :)